runtime/compiler/runtime/RuntimeAssumptions.cpp

```diff
@@ -85,10 +85,10 @@ TR_PatchNOPedGuardSiteOnClassPreInitialize::hashCode(char *sig, uint32_t sigLen)
 void
 TR_PatchNOPedGuardSiteOnClassPreInitialize::reclaim()
    {
-   TR_ASSERT_FATAL(_key != NULL, "Attempt to reclaim an already freed _key");
+   TR_ASSERT_FATAL((void*)_key != NULL, "Attempt to reclaim an already freed _key");
 
    jitPersistentFree((void*)_key);
-   _key = 0;
+   _key = (uintptr_t)(void*)0;
```

Comment:
```suggestion
   _key = 0;
```
`_key` is of type `uintptr_t` so a `0` is fine.

Reply:
See my other comment, but in brief, I believe this is not strictly correct according to the standard, similar to `void *p; memset(&p, 0, sizeof(p));` being an incorrect way to initialize a null pointer to void.